testdata/project-v4-with-plugins/internal/controller/memcached_controller.go

```diff
@@ -167,19 +154,21 @@ func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 				return ctrl.Result{}, err
 			}
 
-			meta.SetStatusCondition(&memcached.Status.Conditions, metav1.Condition{Type: typeDegradedMemcached,
+			changed = meta.SetStatusCondition(&memcached.Status.Conditions, metav1.Condition{Type: typeDegradedMemcached,
 				Status: metav1.ConditionTrue, Reason: "Finalizing",
 				Message: fmt.Sprintf("Finalizer operations for custom resource %s name were successfully accomplished", memcached.Name)})
-
-			if err := r.Status().Update(ctx, memcached); err != nil {
-				log.Error(err, "Failed to update Memcached status")
-				return ctrl.Result{}, err
+			if changed {
+				if err := r.Status().Update(ctx, memcached); err != nil {
+					log.Error(err, "Failed to update Memcached status")
+					return ctrl.Result{}, err
+				}
+				// TODO GET call?
 			}
 
 			log.Info("Removing Finalizer for Memcached after successfully perform the operations")
 			if ok := controllerutil.RemoveFinalizer(memcached, memcachedFinalizer); !ok {
 				log.Error(err, "Failed to remove finalizer for Memcached")
-				return ctrl.Result{Requeue: true}, nil
+				return ctrl.Result{}, fmt.Errorf("Failed to remove finalizer for Memcached")
```

Comment:
Oh, now I think it is clear why we are using requeue(true)
If we change, we will have two texts, which always must be equal with the same info.
So, can we here err = error.NewError(message), log it and use it in the return?

Reply:
created error to avoid text repetition